### PR TITLE
Add single table integration tests

### LINKED ISSUE #1276: Add integration tests for 'Synthesize a Table' demo
### Problem Description

As part of migrating to SDV 1.0, we are creating new demos! Since running the old demos as a workflow caused problems (took a long time and sometimes failed due to jupyter dependencies), we want to make sure that all of the processes in the new demos are made into integration tests.

### Expected behavior

- Make integration tests for all of the `SDV` processes ran in the "Synthesize a Table" demo. There are two of these demos, one for `GaussianCopulaSynthesizer` and one for `CTGANSynthesizer`. Write the tests for both

### Technical Details

- Put the tests in the appropriate files. If you are adding tests that run methods for a `SingleTableSynthesizer`, put it in the file for that synthesizer ie. (`tests/integration/single_table/copulas.py`)
- We don't need to be very strict on the asserts. Just add basic asserts to ensure things aren't broken. For example,
    - Assert sampled data is the right shape
    - Validate the sampled data
    - Make sure any key values in the output are present

### PR DESCRIPTION
**Pull request by frances-h**
Resolve #1276

### CODE REVIEW

**Review by pvk-developer — APPROVED** [review]
I like the structure I may change the other tests that I did to follow the same one 👍🏻

File: tests/integration/single_table/test_copulas.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    assert model_path.exists()
+    assert model_path.is_file()
+    loaded_synthesizer = GaussianCopulaSynthesizer.load(model_path)
+    assert isinstance(synthesizer, GaussianCopulaSynthesizer)
+    assert loaded_synthesizer.get_info() == synthesizer.get_info()
+    assert loaded_synthesizer.metadata.to_dict() == metadata.to_dict()
+    loaded_synthesizer.sample(20)
+
+    # Assert - custom synthesizer
+    assert custom_quality_report.get_score() > 0
```
Comment (amontanez24) [inline tests/integration/single_table/test_copulas.py:111]: Is it ever possible for this to be 0? I'd hope not but just curious

**Review by amontanez24 — APPROVED** [review]
LGTM, very thorough!

File: tests/integration/single_table/test_copulas.py (hunk 1 above)
Comment (frances-h) [reply tests/integration/single_table/test_copulas.py:111]: I think technically yes if every score was zero or `nan` since it's just the average

**Pull request merged**